spot-contracts/contracts/_utils/BondHelpers.sol

```diff
@@ -191,27 +191,33 @@ library BondHelpers {
         return (td, collateralBalances, trancheSupplies);
     }
 
-    /// @notice Given a bond, retrieves the collateral redeemable for
-    ///         each tranche held by the given address.
+    /// @notice Given a bond and a user address computes the tranche amounts proportional to the tranche ratios,
+    //          that can be redeemed for the collateral before maturity.
     /// @param b The address of the bond contract.
     /// @param u The address to check balance for.
-    /// @return The tranche data and an array of collateral balances.
-    function getTrancheCollateralBalances(IBondController b, address u)
+    /// @return The tranche data and an array of tranche token balances.
+    function computeRedeemableTrancheAmounts(IBondController b, address u)
         internal
         view
         returns (TrancheData memory, uint256[] memory)
     {
-        TrancheData memory td;
-        uint256[] memory collateralBalances;
-        uint256[] memory trancheSupplies;
-
-        (td, collateralBalances, trancheSupplies) = getTrancheCollateralizations(b);
+        TrancheData memory td = getTrancheData(b);
+        uint256[] memory redeemableAmts = new uint256[](td.trancheCount);
+
+        // We calculate the minimum value of {trancheBal/trancheRatio} across tranches
+        uint256 min = type(uint256).max;
+        uint8 i;
+        for (i = 0; i < td.trancheCount && min != 0; i++) {
+            uint256 d = (td.tranches[i].balanceOf(u) * TRANCHE_RATIO_GRANULARITY) / td.trancheRatios[i];
+            if (d < min) {
+                min = d;
+            }
```

Comment:
Would it be more efficient to break the min check into a dedicated condition in the loop?
```
if (min == 0) {
  return (td, redeemableAmts);
}
```
Short-circuits, but only if there's a zero balance so depends on the expected context.

Reply:
sure 👍